pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -247,20 +232,27 @@ func (c *Controller) syncOwnedProvisionRequest(ctx context.Context, wl *kueue.Wo
 		attempt := int32(1)
 		shouldCreatePr := false
 		if exists {
-			if isFailed(oldPr) || (isBookingExpired(oldPr) && !workload.IsAdmitted(wl)) {
+			if (isFailed(oldPr) || (isBookingExpired(oldPr) && !workload.IsAdmitted(wl))) &&
 				// if the workload is Admitted we don't want to retry on BookingExpired
+				ac != nil && ac.State == kueue.CheckStatePending {
+				// if the workload is in Retry/Rejected state we don't create another ProvReq
 				attempt = getAttempt(log, oldPr, wl.Name, checkName)
-				if attempt <= c.maxRetries {
-					remainingTime := c.remainingTimeToRetry(oldPr, attempt)
-					if remainingTime <= 0 {
-						shouldCreatePr = true
-						attempt += 1
-					} else if requeAfter == nil || remainingTime < *requeAfter {
-						requeAfter = &remainingTime
-					}
+				if !features.Enabled(features.KeepQuotaForProvReqRetry) {
+					shouldCreatePr = true
+					attempt += 1
+				}
+				remainingTime := c.remainingTimeToRetry(oldPr, attempt, prc)
+				if remainingTime <= 0 {
+					shouldCreatePr = true
+					attempt += 1
+				} else if requeAfter == nil || remainingTime < *requeAfter {
+					requeAfter = &remainingTime
 				}
 			}
 		} else {
+			if oldPr, exists := latestStalePR[checkName]; exists {
```

Comment:
Can you explain / add a comment why this is needed? I'm a bit lost TBH at the moment

Reply:
We need to keep stale ProvReqs so we know what should be the name (that contains retry number) of the new one. At the same time we don't want to process stale ProvReq that are failed and a workload already was retried with backoff - so I created a separate map for ProvReq that shouldn't be processed, but at the same time are need to create new ones

Reply:
but why can't we just reuse activeOrLastPRForChecks